src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+                    polarimetry_dict_raw[ebeam_pos] = data_raw_bg_corr  # dict containing the bg corrected raw data
+
+                    # Calculate the polarimetry results:
+                    # Dictionary keys are (ebeam pos x, ebeam pos y, optional: polarimetry key).
+                    # Note: Takes about 0.25 sec to calc all polarimetry results for one ebeam pos
+                    # and tested on an image of size (256, 1024)
+                    polarimetry_dict_raw.update(
+                        arpolvis.calcPolarimetry(polarimetry_dict_raw, ebeam_pos, size, wl))
+
+                # Convert the data back to polar representation for GUI display.
+                # Note: Takes 0.24 sec to convert one image for display and tested on an image of size (256, 1024)
+                # TODO invalid division check if only bg  # norm so far
+                # TODO move to update image?? as this is the only code we need to reuse
+                #  whenever the legend changes while having the same ebeam pos selected
+                if pos[-1] in ["Ephi", "Etheta", "Ex", "Ey", "Ez"]:  # TODO: hardcoded for now
+                    data = numpy.abs(polarimetry_dict_raw[pos])
+                    order = numpy.log10(data.max())
+                    try:
+                        plotorder = numpy.int(order)
+                    except OverflowError:
+                        # Note: Error is raised, when bg image and image are the same
+                        # (-> cannot convert float infinity to integer)
+                        # TODO is there a better way to handle this problem?
+                        plotorder = 0
+                    polarimetry_dict_raw[pos] = data / 10 ** plotorder
+                    colormap = 2
+                else:
+                    colormap = 1
+                # TODO this gives an e.g. Index/Key error in the logging when failing. why? this code is in a try and except...
+                stokes_data = \
+                    angleres.Rectangular2Polar(polarimetry_dict_raw[pos], colormap=colormap)
+
+                new_md = self._find_metadata(stokes_data.metadata)
+                new_md[model.MD_DIMS] = stokes_data.metadata[model.MD_DIMS] # YXC
+                polarimetry_dict[pos] = model.DataArray(stokes_data, new_md)
+
+                # return an array
+                stokes_data = polarimetry_dict[pos]
+
+            except Exception:
+                logging.exception("Failed to calculate polarimetry results.")
+                return numpy.zeros((1034, 1034, 3))  # return a black image as fallback
+
+        return stokes_data
+
+    def _updateImage(self):
+        """Recomputes the image for the current ebeam position and polarimetry visualization requested."""
+
+        if not self.raw:
+            return
+
+        # TODO check looks like it is called twice when loading a new image/bg image
+        # TODO This was already in the old code - why??, same for in the other projection of AR
```

Comment:
Probably because on the histogram update (but needs to be confirmed). If that's the case, immediately computing the histogram could make sense here, as we are not in such a hurry to show the image anyway (after 6s of computation).

Reply:
I don't understand why calculating the histogram here? You mean before the try statement?

Reply:
Sorry, I didn't mean "here" as in "right here, at that line", but more as in "here, in this class" :-P